src/service/viewport-impl.js

```diff
@@ -286,8 +286,25 @@ export class Viewport {
    * @return {!../layout-rect.LayoutRectDef}}
    */
   getLayoutRect(el) {
-    return this.binding_.getLayoutRect(el,
-        this.getScrollLeft(), this.getScrollTop());
+    const scrollLeft = this.getScrollLeft();
+    const scrollTop = this.getScrollTop();
+
+    // Go up the window hierarchy through friendly iframes.
+    const win = (el.ownerDocument || el).defaultView;
+    if (win && win != this.win_ && getTopWindow(win) == this.win_) {
```

Comment:
That's what `getTopWindow` does. The window has `__AMP_TOP` property and this is what used here. I just hid it behind API.